cs/src/Connections/TunnelConnection.cs

```diff
@@ -364,4 +368,75 @@ internal void OnRetrying(RetryingTunnelConnectionEventArgs e)
     {
         RetryingTunnelConnection?.Invoke(this, e);
     }
+
+    Task<PortForwardRequestMessage> IPortForwardMessageFactory.CreateRequestMessageAsync(int port)
+        => Task.FromResult<PortForwardRequestMessage>(
+            new PortRelayRequestMessage { AccessToken = this.accessToken });
+
+    Task<PortForwardSuccessMessage> IPortForwardMessageFactory.CreateSuccessMessageAsync(int port)
+        => Task.FromResult(new PortForwardSuccessMessage()); // Success messages are not extended.
+
+    Task<PortForwardChannelOpenMessage> IPortForwardMessageFactory.CreateChannelOpenMessageAsync(int port)
+        => Task.FromResult<PortForwardChannelOpenMessage>(
+            new PortRelayConnectRequestMessage { AccessToken = this.accessToken });
+
+    /// <summary>
+    /// Extends port-forward request messagse to include an access token property.
+    /// </summary>
+    private class PortRelayRequestMessage : PortForwardRequestMessage
```

Comment:
Should we have these custom message classes as public contracts to ensure that both the server side and the client side use the same ones?

Reply:
Done.